src/app/components/client/SubscriptionCheck.tsx

```diff
@@ -4,26 +4,19 @@
 
 "use client";
 
-import { getSession, useSession } from "next-auth/react";
+import { useSession } from "next-auth/react";
 import { useEffect } from "react";
-import { hasPremium } from "../../functions/universal/user";
 
 export const SubscriptionCheck = () => {
   const { update } = useSession();
 
   useEffect(() => {
     async function updateSession() {
-      const session = await getSession();
-      if (!hasPremium(session?.user)) {
-        await update();
-      }
+      await update();
     }
 
     void updateSession();
```

Comment:
Given that we're not actually waiting for the Promise to resolve before doing anything, couldn't we just do:

```suggestion
    update();
```

And also I'm assuming you consciously no longer used the empty dependencies array?

Reply:
Yes, but I’ve removed it again in  [8addbcd](https://github.com/mozilla/blurts-server/pull/3654/commits/8addbcda5883bad1899f2af29000ea7fc662f3b6).

Reply:
Regarding awaiting the session update: No, we do not have to wait for it — updated in [1d60a84](https://github.com/mozilla/blurts-server/pull/3654/commits/1d60a844e4568bf0c620d9430ae78d6652a65278).